source/appController/AMDSClientAppController.cpp

```diff
@@ -122,11 +129,20 @@ AMDSClientTCPSocket * AMDSClientAppController::establishSocketConnection(QString
 	return clientTCPSocket;
 }
 
+void AMDSClientAppController::onSocketError(AMDSServer* server, int errorCode, QString socketKey, QString errorMessage)
+{
+	emit socketError(errorCode, server->serverIdentifier(), errorMessage);
```

Comment:
Since you're only using the serverIdentifier, why don't you pass that in the signal rather than the server itself?